### PR TITLE
add release_for_tag and latest_release methods

### PR DESCRIPTION
**Pull request by zqzas**
@pengwynn Thanks!

### CODE REVIEW

File: lib/octokit/client/releases.rb [hunk 1]
```diff
@@ -127,6 +127,26 @@ def delete_release_asset(asset_url, options = {})
         boolean_from_response(:delete, asset_url, options)
       end
 
+      # Get the release for a given tag
+      #
```
Comment (pengwynn) [inline lib/octokit/client/releases.rb]: Could you snip this extra blank line? :scissors:

File: lib/octokit/client/releases.rb [hunk 2]
```diff
@@ -127,6 +127,26 @@ def delete_release_asset(asset_url, options = {})
...
+      # @param repo [Integer, String, Repository, Hash] A GitHub repository
+      # @param tag_name [String] the name for a tag
+      # @return [Sawyer::Resource] The release
+      def release_for_tag(repo, tag_name, options = {})
+        get "#{Repository.path repo}/releases/tags/#{tag_name}", options
+      end
+
+      # Get the latest release
+      #
```
Comment (pengwynn) [inline lib/octokit/client/releases.rb]: And here.

**Comment by pengwynn** [comment]
@zqzas Looks good. A couple of tiny nits before we merge. :sparkles:

**Comment by zqzas** [comment]
@pengwynn gotcha

**Comment by pengwynn** [comment]
Thanks!

**Pull request merged**